libbeat/outputs/elasticsearch/client.go

```diff
@@ -319,7 +331,11 @@ func createEventBulkMeta(
 	}
 
 	if id != "" || version.Major > 7 || (version.Major == 7 && version.Minor >= 5) {
-		return eslegclient.BulkCreateAction{Create: meta}, nil
+		if opType == "" || opType == "create" {
+			return eslegclient.BulkCreateAction{Create: meta}, nil
+		} else if opType == "delete" {
+			return eslegclient.BulkDeleteAction{Delete: meta}, nil
```

Comment:
If I'm following the conditional logic correctly here, we could be returning a `BulkDeleteAction` even if `id == ""`. This would happen when `version.Major > 7 || (version.Major == 7 && version.Minor >= 5)` evaluates to `true`. I don't think that's what we want to do. 

I think we should return a `BulkDeleteAction` if `id != "" && opType == "delete"` so I would just implement this part before  the outer `if` block. And leave the contents of the `if` block as they are currently.